src/amo/reducers/addonsByAuthors.js

```diff
@@ -4,27 +4,31 @@ import invariant from 'invariant';
 
 import { createInternalAddon } from 'core/reducers/addons';
 import type {
-  SearchResultAddonType, ExternalAddonType,
+  AddonType, ExternalAddonType,
 } from 'core/types/addons';
 
 
-type State = {
+export type AddonsByAuthors = {|
   // TODO: It might be nice to eventually stop storing add-ons in this
   // reducer at all and rely on the add-ons in the `addons` reducer.
   // That said, these are partial add-ons returned from the search
   // results and fetching all add-on data for each add-on might be too
   // expensive.
-  byAddonId: { [number]: Array<SearchResultAddonType> },
```

Comment:
I switched because otherwise I got weird flow errors and I couldn't tell it to be either `AddonType | SearchResultType` even though it's valid without errors. I'll try once more but we run `createInternalAddon` on the addon so it should adhere to the type properly. Maybe there's a `createInternalSearchResultAddon` function I need to use?

Reply:
I was wrong; `SearchResultAddonType` here is fine. Restored!